richtextfx/src/main/java/org/fxmisc/richtext/EditableStyledDocument.java

```diff
@@ -19,7 +19,12 @@
 
     ReadOnlyStyledDocument<PS, S> snapshot();
 
-    EventStream<PlainTextChange> plainChanges();
+    default EventStream<PlainTextChange> plainChanges() {
+        return richChanges()
+                // only allow changes that added/removed text; exclude style changes
+                .filter(c -> c.insertedLength() != 0 || c.removedLength() != 0)
+                .map(c -> new PlainTextChange(c.position, c.removed.getText(), c.inserted.getText()));
+    }
```

Comment:
I guess I do care, when I see how much noise 4e6ce6b9 adds in the name of saving one object allocation per change. Also, code comments can go out of sync with code very easily, so self-documenting code is better :)